orchagent/pfcwdorch.cpp

```diff
@@ -334,7 +387,19 @@ void PfcWdOrch<DropHandler, ForwardHandler>::createEntry(const string& key,
         return;
     }
 
-    m_pfcFrameCountersMap.emplace(port.m_port_id, getPfcFrameCounters(port.m_port_id));
+    sai_attribute_t attr;
+    attr.id = SAI_PORT_ATTR_PRIORITY_FLOW_CONTROL;
+
+    sai_status_t status = sai_port_api->get_port_attribute(port.m_port_id, 1, &attr);
```

Comment:
Can we cache the Priority flow control value, and update cache only when it is set? We keep querying this field in PFCWD which is not efficient.

Reply:
It is a valid idea to cache priority, but I think it has to be cached in Port class and in a separate commit. Then all PFC WD can reuse it.

Reply:
let's add a backlog item for this.

Reply:
can you add an issue for this backlog in the repo so that we do not forget this item.

---
In reply to: [168245304](https://github.com/lguohan/sonic-swss/pull/4#discussion_r168245304) [](ancestors = 168245304)

Reply:
https://github.com/Azure/sonic-swss/issues/444